# FLUJO DE CONTROL

El flujo de control ayudará a nuestro programa en la toma de decisiones

<center> <img src='https://4.bp.blogspot.com/_QWwb06N-r1Y/TMeaj_jf7aI/AAAAAAAAARA/CUtpkzy0Pjo/s1600/selectiva.png' width="400" height="250"></center>

## 1. Flujo de control (if - else) 

El uso de condiciones nos permite tener un mayor control sobre el flujo del programa.

In [1]:
# Uso de la sentencia if (si)
if True:
    print('Hola')

Hola


In [2]:
# Sentencia else (si no)
x = 10
if x == 8 :
    print('el valor de x es 8')  # 4 espacios
else:
    print('el valor de x = {} es distinto de 8'.format(x))
    #print('el valor de x es {}'.format(x))

el valor de x = 10 es distinto de 8


In [6]:
# uso de If anidado
a = 5
b = 1
if a == 5:
    print("a vale",a)
    
    if b == 10:
        print('y b vale',b)

a vale 5


## 2. Sentencia elif (sino si)

Se encadena a un if u otro elif para comprobar múltiples condiciones, siempre que las anteriores no se ejecuten:

In [10]:
# Sentencia else (si no)
x=10
if x==8 :
    print('el valor de x es 8')
elif x==9:
    print('el valor de x es 9')
elif x==10:
    print('el valor de x es 10')
else:
    print('el valor de x es distinto de 8,9 y 10')

el valor de x es 10


# EJERCICIOS

#### 1.
Crear un programa que permita decidir a una persona cruzar la calle o no según:
- Si semáforo esta en verde cruzar la calle
- Si semáforo esta en rojo o amarillo no cruzar

La persona debe poder ingresar el estado del semáforo por teclado

In [25]:
semaforo = input('Introducir el estado del semanaforo: ')
semaforo = semaforo.lower()
semaforo

Introducir el estado del semanaforo:  ROJO


'rojo'

In [26]:
if semaforo == 'verde':
    print('puede cruzar la calle')
elif semaforo == 'amarillo' or semaforo == 'rojo':
    print('no cruzar')
else:
    print('valor ingresado incorrectamente')

no cruzar


In [22]:
#
'hola'.upper()

'HOLA'

In [23]:
#
'HOla'.lower()

'hola'

#### 2.
Escribir un programa que pregunte al usuario su edad y muestre por pantalla si es mayor de edad o no.

In [27]:
edad = int(input('Por favor ingrese su edad'))
edad

Por favor ingrese su edad 15


15

In [28]:
if edad >=18:
    print(f'Es mayor de edad y tiene {edad}')
else:
    print('es menor de edad')

es menor de edad


#### 3.
Escribir un programa que almacene la cadena de caracteres <b>contraseña</b> en una variable, pregunte al usuario por la contraseña e imprima por pantalla si la contraseña introducida por el usuario coincide con la guardada en la variable sin tener en cuenta mayúsculas y minúsculas.

### 4. 
Escribir un programa que pida al usuario un número entero y muestre por pantalla si es par o impar.


#### 5.
Los tramos impositivos para la declaración de la renta en un determinado país son los siguientes:

<table>
    <thead>
        <th style="text-align: center">Renta</th>
        <th style="text-align: center">% de Impuesto</th>
    </thead>
    <tbody>
        <tr>
            <td style="text-align: center">Menos de 10000€</td>
            <td style="text-align: center">5%</td>
        </tr>
        <tr>
            <td style="text-align: center">Entre 10000€ y 20000€</td>
            <td style="text-align: center">15%</td>
        </tr>
        <tr>
            <td style="text-align: center">Entre 20000€ y 35000€</td>
            <td style="text-align: center">20%</td>
        </tr>
        <tr>
            <td style="text-align: center">Entre 35000€ y 60000€</td>
            <td style="text-align: center">30%</td>
        </tr>
        <tr>
            <td style="text-align: center">Más de 60000€</td>
            <td style="text-align: center">45%</td>
        </tr>
    </tbody>
</table>


Realizar un programa que pueda decir el % de impuestos que una persona deba pagar según su sueldo

#### 6. 
Realiza un programa que lea dos números por teclado y permita elegir entre 3 opciones en un menú:

- Mostrar una suma de los dos números
- Mostrar una resta de los dos números (el primero menos el segundo)
- Mostrar una multiplicación de los dos números
- En caso de introducir una opción inválida, el programa informará de que no es correcta.


#### 7.
La pizzería Bella Napoli ofrece pizzas vegetarianas y no vegetarianas a sus clientes. Los ingredientes para cada tipo de pizza aparecen a continuación.

- Ingredientes vegetarianos: Pimiento y tofu.
- Ingredientes no vegetarianos: Peperoni, Jamón y Salmón.

Escribir un programa que pregunte al usuario si quiere una pizza vegetariana o no, y en función de su respuesta le muestre un menú con los ingredientes disponibles para que elija. Solo se puede eligir un ingrediente además de la mozzarella y el tomate que están en todas la pizzas. Al final se debe mostrar por pantalla si la pizza elegida es vegetariana o no y todos los ingredientes que lleva.